### PR TITLE
Feature/sg 255 add class for supported strings

### PR DESCRIPTION
**Pull request by Louis-Dupont**
Introduce a breaking change since we rename the segmentation transorms.
The recipes using these transforms were tested and worked

### CODE REVIEW

File: src/super_gradients/training/utils/callbacks.py [hunk 1]
```diff
@@ -765,3 +768,26 @@ def __init__(self, lr_placeholder):
...
+CALLBACKS = {
+    CallbackNames.DECI_LAB_UPLOAD: DeciLabUploadCallback,
+    CallbackNames.LR_CALLBACK_BASE: LRCallbackBase,
+    CallbackNames.LR_SCHEDULER: LRSchedulerCallback,
+    CallbackNames.METRICS_UPDATE: MetricsUpdateCallback,
+    CallbackNames.MODEL_CONVERSION_CHECK: ModelConversionCheckCallback,
+    CallbackNames.EARLY_STOP: EarlyStop,
+    CallbackNames.DETECTION_MULTISCALE_PREPREDICTION: DetectionMultiscalePrePredictionCallback,
+    CallbackNames.YOLOX_TRAINING_STAGE_SWITCH: YoloXTrainingStageSwitchCallback
+}
```
Comment (Louis-Dupont) [inline src/super_gradients/training/utils/callbacks.py:793]: I did the same on callbacks to have a unified approach but I got an error when importing EarlyStop in the CI.
I think this is due to circular import, and to avoid it I think we should do like with optimizers, having a folder including 2 modules, one with the functions and one with what I just added.

What do you think ? Should we just leave this for later and not modify the callback factory, or should we do the refacto I talked about ? Or any other option ?

Comment (Louis-Dupont) [reply src/super_gradients/training/utils/callbacks.py:793]: https://github.com/Deci-AI/super-gradients/blob/bfaaefbc25003f0abc97186aa561d53b8471d47f/src/super_gradients/training/utils/callbacks.py#L773-L793

File: src/super_gradients/examples/SG_quickstart_segmentation.ipynb [hunk 2]
```diff
@@ -187,13 +187,13 @@
         "                  \"batch_size\": 32,\n",
         "                  \"val_batch_size\":32,\n",
         "                  \"image_mask_transforms_aug\": transforms.Compose(\n",
-        "                      [ColorJitterSeg(brightness=0.5, contrast=0.5, saturation=0.5),\n",
+        "                      [SegColorJitter(brightness=0.5, contrast=0.5, saturation=0.5),\n",
```
Comment (shaydeci) [inline src/super_gradients/examples/SG_quickstart_segmentation.ipynb:190]: The other transforms here are also supposed to be renamed.

File: src/super_gradients/examples/SG_transfer_learning_semantic_segmentation.ipynb [hunk 3]
```diff
@@ -183,13 +183,13 @@
     "                  \"batch_size\": 32,\n",
     "                  \"val_batch_size\":32,\n",
     "                  \"image_mask_transforms_aug\": transforms.Compose(\n",
-    "                      [ColorJitterSeg(brightness=0.5, contrast=0.5, saturation=0.5),\n",
+    "                      [SegColorJitter(brightness=0.5, contrast=0.5, saturation=0.5),\n",
```
Comment (shaydeci) [inline src/super_gradients/examples/SG_transfer_learning_semantic_segmentation.ipynb:186]: The other transforms here should also be renamed.

File: docs/searchindex.js [hunk 4]
```diff
@@ -1 +1 @@
...
...thon:[],pytorch:37,question:37,quick:37,quickli:37,r_squared_loss:30,raffl:37,recip:36,regnet:32,regularization_util:34,reproduc:36,repvgg:32,resnet18:37,resnet:32,resnext:32,rmsprop_tf:35,s3_connector:16,s:[10,37],script:36,segment:37,segmentation_dataset:27,segmentation_metr:31,segmentation_util:34,semant:37,senet:32,set:36,sg:37,sg_dataset:23,sg_model:[22,33],sg_model_except:28,sg_model_util:34,sg_modul:32,shelfnet:32,shelfnet_ohem_loss:30,shelfnet_semantic_encoding_loss:30,shufflenet:32,shufflenetv2:32,singleton:20,soon:37,sota:37,ssd:32,ssd_loss:30,ssd_util:34,stabl:[],start:37,style:0,submodul:[13,14,15,16,17,19,20,21,23,25,26,27,28,29,30,31,32,33,34,35],subpackag:[18,23,34],super_gradi:[2,3,4,5,6,7,8,9,11,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35],supergradi:[10,36,37],support:36,survei:37,swag:37,tabl:[10,37],task:36,technic:10,those:37,tool:36,train:[22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37],transfer:37,type:36,up:36,us:37,user:10,util:[22,29,34,35],version:[],vgg:32,vision:37,walkthrough:36,we:37,weight:[36,37],weight_averaging_util:34,welcom:10,what:[36,37],who:37,why:37,yolo_v3_loss:30,yolo_v5_loss:30,yolov3:32,yolov5:32,your:[36,37]}})
```
Comment (shaydeci) [inline docs/searchindex.js:1]: Whats all this ?

File: src/super_gradients/training/utils/callbacks/all_callbacks.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    CallbackNames.METRICS_UPDATE: MetricsUpdateCallback,
+    CallbackNames.MODEL_CONVERSION_CHECK: ModelConversionCheckCallback,
+    CallbackNames.EARLY_STOP: EarlyStop,
+    CallbackNames.DETECTION_MULTISCALE_PREPREDICTION: DetectionMultiscalePrePredictionCallback,
+    CallbackNames.YOLOX_TRAINING_STAGE_SWITCH: YoloXTrainingStageSwitchCallback
+}
+
+
+# DICT FOR LEGACY LR HARD-CODED REGIMES, WILL BE DELETED IN THE FUTURE
+class LRScheduler:
```
Comment (shaydeci) [inline src/super_gradients/training/utils/callbacks/all_callbacks.py:34]: Not consistent with your logic of adding the names suffix, can be confusing.

**Review by shaydeci — CHANGES_REQUESTED** [review]
inline.

File: src/super_gradients/training/metrics/all_metrics.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+from super_gradients.training.metrics import Accuracy, Top5, DetectionMetrics, IoU, PixelAccuracy, BinaryIOU, Dice,\
+    BinaryDice
+
+
+class MetricNames:
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/training/metrics/all_metrics.py]: Changed here: e53d29780f25ec1cd250d6a50f3a2f1eb0ab2a9a
Not sure about the name as well but I cant think of a term other than object to categorize all these classes (loss, metrics, lrscheduler, optimizer, ... )

File: src/super_gradients/training/models/classification_models/beit.py [hunk 7]
```diff
@@ -36,7 +36,7 @@
 
 
 def resize_pos_embed(posemb, posemb_new, num_tokens=1, gs_new=()):
-    # Rescale the grid of position embeddings when loading from state_dict. Adapted from
+    # SegRescale the grid of position embeddings when loading from state_dict. Adapted from
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/training/models/classification_models/beit.py]: removed

File: src/super_gradients/examples/SG_transfer_learning_semantic_segmentation.ipynb (hunk 3 above)
Comment (Louis-Dupont) [reply src/super_gradients/examples/SG_transfer_learning_semantic_segmentation.ipynb]: I removed this change anyway because the notebooks were changed since last time

File: docs/searchindex.js (hunk 4 above)
Comment (Louis-Dupont) [reply docs/searchindex.js]: I am not sure, I removed the change

File: src/super_gradients/training/utils/callbacks/all_callbacks.py (hunk 5 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/utils/callbacks/all_callbacks.py]: Also changed in e53d29780f25ec1cd250d6a50f3a2f1eb0ab2a9a

File: src/super_gradients/examples/SG_quickstart_segmentation.ipynb (hunk 2 above)
Comment (Louis-Dupont) [reply src/super_gradients/examples/SG_quickstart_segmentation.ipynb]: I removed this change anyway because the notebooks were changed since last time

File: src/super_gradients/recipes/dataset_params/cityscapes_stdc_seg50_dataset_params.yaml [hunk 8]
```diff
@@ -5,28 +5,28 @@ defaults:
 train_dataset_params:
   transforms:
     # for more options see common.factories.transforms_factory.py
-    - ColorJitterSeg:
+    - SegColorJitter:
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/recipes/dataset_params/cityscapes_stdc_seg50_dataset_params.yaml:8]: All the detection Names and classes start with "Detection", while our segmentation names finish with Seg.
Since I had to rename all the segmentation transforms, I thought that it would be better to follow the same convention as with Detection. And because we want the names and transforms to match, I also changed the names

**Review by shaydeci — APPROVED** [review]
LGTM
